multiplexer/main.go

```diff
@@ -13,33 +19,135 @@ type URLs struct {
 	List []string `json:"list"`
 }
 
+//Client is a http clieng
+type Client struct {
+	client *http.Client
+}
+
+//Response is the response sent to the client
+type Response struct {
+	index   int
+	content []byte
+}
+
+//Job is a single url with its index
+type Job struct {
+	index int
+	site  string
+}
+
+func newClient() *Client {
+	return &Client{
+		client: &http.Client{
+			Timeout: time.Second,
+		},
+	}
+}
+
+func (c Client) getContent(url string) ([]byte, error) {
+	resp, err := c.client.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return b, nil
+}
+
 func main() {
 	var urls URLs
+	var jwg sync.WaitGroup
```

Comment:
You need to init urls var, two wgs, jobs channel and result channel in your handler. If you will have more that one client, you will get races.
And this is why you have a panics on second request